diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -193,8 +193,31 @@ namespace diff_drive_controller{
     controller_nh.param("angular/z/max_acceleration"       , limiter_ang_.max_acceleration       ,  limiter_ang_.max_acceleration      );
     controller_nh.param("angular/z/min_acceleration"       , limiter_ang_.min_acceleration       , -limiter_ang_.max_acceleration      );
 
-    if (!setOdomParamsFromUrdf(root_nh, left_wheel_names[0], right_wheel_names[0]))
-      return false;
+    // If either parameter is not available, we need to look up the value in the URDF
+    bool lookup_wheel_separation = !controller_nh.getParam("wheel_separation", wheel_separation_);
+    bool lookup_wheel_radius = !controller_nh.getParam("wheel_radius", wheel_radius_);
+
+    // Short-circuit in case we don't need to look up anything, so we don't have to parse the URDF
+    if(lookup_wheel_separation or lookup_wheel_radius)
```

Comment:
But the actual odometry values are now being set [here](https://github.com/bmagyar/ros_controllers/blob/separation_and_diameter_params/diff_drive_controller/src/diff_drive_controller.cpp#L217). This is obviously a minor point, I just want to make sure the function name doesn't confuse future developers :)